packages/lab/src/__tests__/__e2e__/overlay/Overlay.cy.tsx

```diff
@@ -32,26 +32,28 @@ describe("GIVEN an Overlay", () => {
     });
 
     it("THEN it should remain open until outside Overlay click or close button click", () => {
-      const closeSpy = cy.stub().as("closeSpy");
-      cy.mount(<Default onClose={closeSpy} />);
+      const onOpenChangeSpy = cy.stub().as("onOpenChangeSpy");
+      cy.mount(<Default onOpenChange={onOpenChangeSpy} />);
 
       cy.findByRole("button", { name: /Show Overlay/i }).realClick();
       cy.findByRole("dialog").should("be.visible");
+      cy.get("@onOpenChangeSpy").should("have.callCount", 1);
       cy.findByRole("button", { name: /Close Overlay/i }).realClick();
-      cy.get("@closeSpy").should("have.callCount", 1);
       cy.findByRole("dialog").should("not.exist");
+      cy.get("@onOpenChangeSpy").should("have.callCount", 2);
 
       cy.findByRole("button", { name: /Show Overlay/i }).realClick();
       cy.findByRole("dialog").should("be.visible");
-      // dbl click to trigger outside of Overlay click, see SOF post
-      // https://stackoverflow.com/questions/51254946/cypress-does-not-always-executes-click-on-element/58302240#58302240
```

Comment:
looks like this behaviour has been fixed - https://github.com/cypress-io/cypress/pull/3030